### PR TITLE
[SYCLomatic] Add warning message when the option for Intel's Extensions for Device Information is on

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

File: clang/test/dpct/memory_management_restricted.cu [hunk 1]
```diff
@@ -380,13 +380,35 @@ void foobar() {
 
   size_t free_mem, total_mem;
 
-  //CHECK: dpct::get_current_device().get_memory_info(free_mem, total_mem);
+  // CHECK:  /*
+  // CHECK-NEXT:DPCT1106:{{[0-9]+}}: 'cudaMemGetInfo' was migrated with Intel's Extensions for Device Information which may be not supported by all compilers or runtimes. You may need to adjust the code.
+  // CHECK-NEXT:*/
+  // CHECK-NEXT:dpct::get_current_device().get_memory_info(free_mem, total_mem);
   cudaMemGetInfo(&free_mem, &total_mem);
```
Comment (wenhuiNi) [inline clang/test/dpct/memory_management_restricted.cu:387]: I think we also need update diagnostic information in APINames.inc

File: clang/test/dpct/memory_management_restricted.cu [hunk 2]
```diff
@@ -380,13 +380,35 @@ void foobar() {
...
+  // CHECK-NEXT:*/
+  // CHECK-NEXT: /*
+  // CHECK-NEXT:DPCT1106:{{[0-9]+}}: 'cudaMemGetInfo' was migrated with Intel's Extensions for Device Information which may be not supported by all compilers or runtimes. You may need to adjust the code.
+  // CHECK-NEXT:*/
+  // CHECK-NEXT:  MY_ERROR_CHECKER((dpct::get_current_device().get_memory_info(free_mem, total_mem), 0));
   MY_ERROR_CHECKER(cudaMemGetInfo(&free_mem, &total_mem));
-  //CHECK: dpct::get_current_device().get_memory_info(free_mem, total_mem);
+
+  // CHECK:  /*
+  // CHECK-NEXT:  DPCT1106:{{[0-9]+}}: 'cuMemGetInfo_v2' was migrated with Intel's Extensions for Device Information which may be not supported by all compilers or runtimes. You may need to adjust the code.
```
Comment (zhimingwang36) [inline clang/test/dpct/memory_management_restricted.cu]: cuMemGetInfo() should be used,  as cuMemGetInfo_v2 is not availalbe in test code.

File: clang/test/dpct/memory_management_restricted.cu (hunk 1 above)
Comment (tomflinda) [reply clang/test/dpct/memory_management_restricted.cu:387]: Okay

File: clang/test/dpct/memory_management_restricted.cu (hunk 2 above)
Comment (tomflinda) [reply clang/test/dpct/memory_management_restricted.cu]: Okay

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**